File: Decorator/WeaponSample/src/Gun.java
public class Gun implements IWeapon {
    private String _name;

    public Gun(String _name) {
        this._name = _name;
    }

    @Override
    public void Fire() {
        System.out.println(this._name + " fired a round, Attachments:");
    }

    @Override
    public String GetName() {
        return this._name;
    }

    @Override
    public void SetName(String name) {
        this._name = name;
    }
}

File: Decorator/csharp/Sample/Decorator/Weapon/Gun.cs
using System;

namespace Decorator.Weapon
{
    public class Gun : IWeapon
    {
        public Gun(string weaponName)
        {
            Name = weaponName;
        }
        public void Fire()
        {
            Console.WriteLine($"{Name} fired a round, Attachments:");
        }

        public string Name { get; }
    }
}